### PR TITLE
Corrected Matk status calculation

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: https://github.com/rathena/rathena/issues/8705

* **Server Mode**: renewal

* **Description of Pull Request**:

### CODE REVIEW

File: src/map/status.cpp [hunk 1]
```diff
@@ -6027,6 +6027,18 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
 		status->matk_max = status_base_matk_max(status) + (sd?sd->bonus.ematk:0);
+
+		if (sd != nullptr && sd->matk_rate != 100) {
+			status->matk_max = status->matk_max * sd->matk_rate/100;
+			status->matk_min = status->matk_min * sd->matk_rate/100;
+		}
+
+		if ((bl.type == BL_HOM && battle_config.hom_setting&HOMSET_SAME_MATK)  /// Hom Min Matk is always the same as Max Matk
+				|| (sc && sc->getSCE(SC_RECOGNIZEDSPELL)))
+			status->matk_min = status->matk_max;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6038]: Please try to keep the code/formatting the same between renewal and prerenewal.
Suggest to copy lines 6126-6130 here.

File: src/map/status.cpp [hunk 2]
```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
+		if (sd != nullptr) {
+			status->matk_min = status_calc_ematk(&bl, sc, status->matk_min);
+			status->matk_max = status->matk_min;
+
+			// Adds weapon magic attack (wMATK) modifications
+			// This is the only portion in MATK that varies depending on the weapon level and refinement rate.
+			if (b_status->lhw.matk) {
+				//sd->state.lr_flag = 1; //?? why was that set here
+				status->lhw.matk = b_status->lhw.matk;
+				sd->state.lr_flag = 0;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6060]: Is this intended? See commented line above.

File: src/map/status.cpp [hunk 3]
```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
-				}
+			if (b_status->rhw.matk) {
+				status->rhw.matk = b_status->rhw.matk;
+			}
 
-				if (b_status->rhw.matk) {
-					status->rhw.matk = b_status->rhw.matk;
-				}
+			int wMatk = 0;
+			int variance = 0;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6068]:
```suggestion
			int32 wMatk = 0;
			int32 variance = 0;
```

File: src/map/status.cpp [hunk 4]
```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
+		if (sc && sc->getSCE(SC_MAGICPOWER)) {
+			status->matk_min += status->matk_min * sc->getSCE(SC_MAGICPOWER)->val3/100;
+			status->matk_max += status->matk_max * sc->getSCE(SC_MAGICPOWER)->val3/100;
 		}
 
-		if ((bl.type == BL_HOM && battle_config.hom_setting&HOMSET_SAME_MATK)  /// Hom Min Matk is always the same as Max Matk
-				|| (sc && sc->getSCE(SC_RECOGNIZEDSPELL)))
-			status->matk_min = status->matk_max;
+		if (sd != nullptr) {
+			uint16 skill_lv;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6092]:
```suggestion
```

File: src/map/status.cpp [hunk 5]
```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
+			if (sd->bonus.ematk > 0) {
+				status->matk_min += sd->bonus.ematk;
+				status->matk_max += sd->bonus.ematk;
+			}
+
+			if (pc_checkskill(sd, SU_POWEROFLAND) > 0 && pc_checkskill_summoner(sd, SUMMONER_POWER_LAND) >= 20) {
+				status->matk_min += status->matk_min * 20 / 100;
+				status->matk_max += status->matk_max * 20 / 100;
+			}
+			if ((skill_lv = pc_checkskill(sd, NV_TRANSCENDENCE)) > 0) {
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6102]:
```suggestion

			if (uint16 skill_lv = pc_checkskill(sd, NV_TRANSCENDENCE); skill_lv > 0) {
```

File: src/map/status.cpp [hunk 6]
```diff
@@ -6036,76 +6048,87 @@ void status_calc_bl_main(struct block_list& bl, std::bitset<SCB_MAX> flag)
...
-						status->matk_min += 15 * skill_lv + (skill_lv > 4 ? 25 : 0);
-				}
+		if (sd != nullptr) {
+			status->matk_min = status_calc_ematk(&bl, sc, status->matk_min);
+			status->matk_max = status->matk_min;
+
+			// Adds weapon magic attack (wMATK) modifications
+			// This is the only portion in MATK that varies depending on the weapon level and refinement rate.
+			if (b_status->lhw.matk) {
+				//sd->state.lr_flag = 1; //?? why was that set here
```
Comment (Lemongrass3110) [inline src/map/status.cpp:6058]: Please consider removing.

File: src/map/status.cpp (hunk 2 above)
Comment (Atemo) [reply src/map/status.cpp:6060]: It looks like a leftover of something, the flag is unused here

**Pull request merged**